addons/web/static/src/views/fields/char/char_field.js

```diff
@@ -53,13 +57,50 @@ export class CharField extends Component {
             isPassword: this.props.isPassword,
         });
     }
+    get hasDynamicPlaceholder() {
+        return this.props.dynamicPlaceholder && !this.props.readonly;
+    }
 
     parse(value) {
         if (this.shouldTrim) {
             return value.trim();
         }
         return value;
     }
+
+    onBlur() {
+        this.selectionStart = this.input.el.selectionStart;
+    }
+
+    async onDynamicPlaceholderOpen() {
+        await this.dynamicPlaceholder.open({
+            validateCallback: this.onDynamicPlaceholderValidate.bind(this),
+        });
+    }
+
+    async onDynamicPlaceholderValidate(chain, defaultValue) {
+        if (chain) {
+            this.input.el.focus();
+            defaultValue = defaultValue.replace("|||", "");
+            let dynamicPlaceholder =
+                "object." +
+                chain +
+                (defaultValue && defaultValue !== "" ? ` ||| ${defaultValue}` : "");
+            dynamicPlaceholder = `{{${dynamicPlaceholder}}}`;
+            const newValue =
+                this.input.el.value.slice(0, this.selectionStart) +
+                dynamicPlaceholder +
+                this.input.el.value.slice(this.selectionStart);
+
+            this.input.el.value = newValue;
+            // trigger events to make the field dirty
+            this.input.el.dispatchEvent(new InputEvent("input"));
+            this.input.el.dispatchEvent(new KeyboardEvent("keydown"));
+            this.input.el.focus();
+            const newCursorPos = this.selectionStart + dynamicPlaceholder.length;
+            this.input.el.setSelectionRange(newCursorPos, newCursorPos);
```

Comment:
Could you try:
```suggestion
            this.input.el.setRangeText(dynamicPlaceholder, this.selectionStart, this.selectionStart, "end");
```
To see if you have the same behavior? [Source](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setRangeText)